ribs/schedulers/_scheduler.py

```diff
@@ -75,6 +83,8 @@ def __init__(self, archive, emitters, add_mode="batch"):
         self._emitters = emitters
         self._add_mode = add_mode
 
+        self._result_archive = result_archive
```

Comment:
Add a check that ensures that `id(result_archive) != id(archive)` -- this is in case someone passes in the same archive for `result_archive` and `archive`